staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -356,5 +357,7 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 	if a.onRotate != nil && oldCreds != nil && !reflect.DeepEqual(oldCreds.cert, a.cachedCreds.cert) {
 		a.onRotate()
 	}
+	metrics.ClientCertRotationAge.Observe(oldCreds.cert.Leaf.NotAfter.Sub(oldCreds.cert.Leaf.NotBefore))
```

Comment:
can't oldCreds be nil here?

Reply:
should probably be in the if block above so if the same cert is returned (we didn't actually rotate), we don't record a rotation:

```go
if oldCreds != nil && !reflect.DeepEqual(oldCreds.cert, a.cachedCreds.cert) {
  metrics.ClientCertRotationAge.Observe(oldCreds.cert.Leaf.NotAfter.Sub(oldCreds.cert.Leaf.NotBefore))
  if a.onRotate != nil {
    a.onRotate()
  }
}
```

Reply:
Ah, good catch. Fixed.

Reply:
the prometheus metric wired to this says "Count of the number of seconds the last client certificate lived before being rotated.", but this looks like it is reporting the valid age of the cert. To match the prometheus docs, I expected something like `time.Now().Sub(oldCreds.cert.Leaf.NotBefore)`